crates/bitcoin-da/src/service.rs

```diff
@@ -560,9 +560,25 @@ impl BitcoinService {
     #[instrument(level = "trace", skip_all, ret)]
     pub async fn get_fee_rate_as_sat_vb(&self) -> Result<u64, anyhow::Error> {
         // If network is regtest or signet, mempool space is not available
-        let smart_fee = match get_fee_rate_from_mempool_space(self.network).await? {
-            Some(fee_rate) => Some(fee_rate),
-            None => self.client.estimate_smart_fee(1, None).await?.fee_rate,
+        let smart_fee = match get_fee_rate_from_mempool_space(self.network).await {
+            Ok(Some(fee_rate)) => Some(fee_rate),
+            Err(e) => {
```

Comment:
why not Err(e) | Ok(None) here?

Reply:
In order to log the error I need to use the inner value of Err, in that case I cannot use the inner value, it raises a compiler error saying sth like it may not be Err so you cannot use the inner value